common/include/edgeai_demo_config.h

```diff
@@ -0,0 +1,482 @@
+ *
+ *    Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ *    Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *    Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef _TI_EDGEAI_TIOVX_DEMO_CONFIG_H_
+#define _TI_EDGEAI_TIOVX_DEMO_CONFIG_H_
+
+/* Standard headers. */
+#include <map>
+#include <set>
+
+/* Third-party headers. */
+#include <yaml-cpp/yaml.h>
+
+/**
+ * \defgroup group_edgeai_demo_config Demo configuration processing.
+ *
+ * \brief Core demo configuration logic for parsing the configuration
+ *        YAML file and setting up complex data flows.
+ *
+ * \ingroup group_edgeai_common
+ */
+
+namespace ti::edgeai::common
+{
+#define DEMO_CONFIG_APPSINK_BUFF_DEPTH  (2)
+
+    using namespace std;
+    using namespace ti::edgeai::common;
+
+    struct SubFlowConfig
+    {
+        string                 model;
+        string                 output;
+        vector<int>            mosaic_info;
+    };
+
+    struct FlowConfig
+    {
+        string                input;
+        vector<SubFlowConfig> subflow_configs;
+    };
+
+    /* Forward declaration. */
+    class OutputInfo;
+
+    /**
+     * \brief Class for holding mosaic parameters for constructing the output.
+     *
+     * \ingroup group_edgeai_demo_config
+     */
+    class MosaicInfo
+    {
+        public:
+            /** Default constructor. Use the compiler generated default one. */
+            MosaicInfo();
+
+            /** Constructor.
+             *
+             * @param data Integer Vector containing posX,posY,width,height
+             */
+            MosaicInfo(vector<int> data);
+
+            /** Destructor. */
+            ~MosaicInfo();
+
+        public:
+            void dumpInfo(const char *prefix="") const;
```

Comment:
If you have 2 groups, is the intent that one group is for methods/functions, and another for variables, then this should be moved up to first public section.

Reply:
Updated this.